verification/src/transaction_verifier.rs

```diff
@@ -140,6 +153,50 @@ impl<'a> EmptyVerifier<'a> {
     }
 }
 
+pub struct MaturityVerifier<'a> {
+    transaction: &'a Transaction,
+    tip_number: BlockNumber,
+    cell_set: &'a FnvHashMap<H256, TransactionMeta>,
+    cellbase_maturity: usize,
+}
+
+impl<'a> MaturityVerifier<'a> {
+    pub fn new(
+        transaction: &'a Transaction,
+        tip_number: BlockNumber,
+        cell_set: &'a FnvHashMap<H256, TransactionMeta>,
+        cellbase_maturity: usize,
+    ) -> Self {
+        MaturityVerifier {
+            transaction,
+            tip_number,
+            cell_set,
+            cellbase_maturity,
+        }
+    }
+
+    pub fn verify(&self) -> Result<(), TransactionError> {
+        let immature_spend = self.transaction.inputs().iter().any(|input| {
+            match self.cell_set.get(&input.previous_output.hash) {
```

Comment:
You can get `CellMeta` from `ResolvedRransaction#inputs`.
https://github.com/nervosnetwork/ckb/blob/develop/core/src/cell.rs#L10

Add `is_cellbase` to CellMeta, and use CellMeta to instead the directly reference to cell_set and transaction_meta

Reply:
I see. Maybe I need to add a field called `is_cellbase` for `CellMeta`.